src/core/lib/surface/call.c

```diff
@@ -1087,6 +1089,24 @@ static void receiving_initial_metadata_ready(grpc_exec_ctx *exec_ctx,
         &call->metadata_batch[1 /* is_receiving */][0 /* is_trailing */];
     grpc_metadata_batch_filter(md, recv_initial_filter, call);
 
+    /* make sure the received grpc-encoding is amongst the ones listed in
+     * grpc-accept-encoding */
+
+    GPR_ASSERT(call->encodings_accepted_by_peer != 0);
```

Comment:
@dgquintas : Thanks for clarifying this offline. 
Just to document what we talked:  The LSB is always set either at line 265 (by the server when creating the call object) and may also be set in `set_encodings_accepted_by_peer()` function which is called (in some cases) by the `grpc_metadata_batch_filter()` function in the previous line (i.e line: 1090)